src/components/actions/evaluatorFunctions.ts

```diff
@@ -8,6 +8,29 @@ Any changes done here should also be replicated in front-end
 
 import { DateTime, Duration } from 'luxon'
 
+interface FilterOptions {
+  key?: string
+  rule?: 'exclude' | 'include'
+  values: string | string[]
+}
+
+const filterArray = (valuesArray: unknown[], options: FilterOptions) => {
+  const { key, rule = 'exclude', values } = options
+  const compareValues = Array.isArray(values) ? values : [values]
+  return valuesArray.filter((element) => {
```

Comment:
`typeof element === 'object'` isn't *quite* enough as both `null` and arrays return true.

It'll need to be `typeof element === 'object' && !Array.isArray(element) && element !== null`

Feel free to abstract this out into a `isObject` utility function if you like.

Reply:
Hmmm no problem. I'll do this now